### PR TITLE
prevent multiple orchestrations at the same time

### PR DESCRIPTION
**Pull request by MaximilianMeister**
Signed-off-by: Maximilian Meister <mmeister@suse.de>

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
[How to apply the changes?](https://trello.com/c/O2BGEKMv/688-how-to-apply-the-changes)

File: app/models/orchestration.rb [hunk 1]
```diff
@@ -26,9 +27,14 @@ def run
   # rubocop:enable Rails/SkipsModelValidations
 
   def self.run(kind: :bootstrap)
+    raise OrchestrationOngoing unless runnable?
     Orchestration.create!(kind: kind).tap(&:run)
   end
 
+  def self.runnable?
+    Orchestration.all.none?(&:in_progress?)
```
Comment (ereslibre) [inline app/models/orchestration.rb]: It's always better to use the scopes:

```
irb(main):005:0> Orchestration.all.none?(&:in_progress?)
  Orchestration Load (0.4ms)  SELECT `orchestrations`.* FROM `orchestrations`
=> true
```

as opposed to

```
irb(main):004:0> Orchestration.in_progress.empty?
   (3.5ms)  SELECT COUNT(*) FROM `orchestrations` WHERE `orchestrations`.`status` = 0
=> true
```

With `Orchestration.in_progress.empty?` we don't need to load the records in memory just to check if there's any, and the SQL generated sentence is smarter.

**Review by ereslibre — CHANGES_REQUESTED** [review]
Except for the comment, LGTM.

File: app/models/orchestration.rb (hunk 1 above)
Comment (MaximilianMeister) [reply app/models/orchestration.rb]: updated, thx

**Review by ereslibre — APPROVED** [review]
LGTM

**Review by vitoravelino — APPROVED** [review]
:+1:

**Pull request merged**